File: 백준/Silver/10828. 스택/스택.go
package main

import (
	"bufio"
	"os"
	"strconv"
)

var (
	sc    *bufio.Scanner
	bw    *bufio.Writer
	stack []int
)

func init() {
	sc = bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	bw = bufio.NewWriter(os.Stdout)
}

func scanInt() int {
	sc.Scan()
	n, _ := strconv.Atoi(sc.Text())
	return n
}

func main() {
	defer bw.Flush()
	N := scanInt()

	for i := 0; i < N; i++ {
		sc.Scan()
		order := sc.Text()
		switch order {
		case "push":
			num := scanInt()
			stack = append(stack, num)
		case "pop":
			if len(stack) == 0 {
				bw.WriteString("-1\n")
			} else {
				out := stack[len(stack)-1]
				stack = stack[:len(stack)-1]
				bw.WriteString(strconv.Itoa(out) + "\n")
			}
		case "size":
			bw.WriteString(strconv.Itoa(len(stack)) + "\n")
		case "empty":
			if len(stack) == 0 {
				bw.WriteString("1\n")
			} else {
				bw.WriteString("0\n")
			}
		case "top":
			if len(stack) == 0 {
				bw.WriteString("-1\n")
			} else {
				bw.WriteString(strconv.Itoa(stack[len(stack)-1]) + "\n")
			}
		}

	}
}

File: 백준/Silver/10828. 스택/스택.cc
#include "bits/stdc++.h"

using namespace std;

int main() {
    ios::sync_with_stdio(false), cin.tie(nullptr);
    stack<int> stack1;

    int N;
    cin >> N;
    for (int i = 0; i < N; ++i) {
        string cmd;
        cin >> cmd;

        if (cmd == "push") {
            int num;
            cin >> num;
            stack1.push(num);
        } else if (cmd == "pop") {
            if (stack1.empty()) {
                cout << -1 << "\n";
            } else {
                cout << stack1.top() << "\n";
                stack1.pop();
            }
        } else if (cmd == "size") {
            cout << stack1.size() << "\n";
        } else if (cmd == "empty") {
            if (stack1.empty()) {
                cout << 1 << "\n";
            } else {
                cout << 0 << "\n";
            }
        } else if (cmd == "top") {
            if (stack1.empty()) {
                cout << -1 << "\n";
            } else {
                cout << stack1.top() << "\n";
            }
        }
    }

    return 0;
}